### PR TITLE
Add tests for GDTCORPlatform

### PR DESCRIPTION
**Pull request by doudounan**
Add tests for GDTCORPlatform.

### CODE REVIEW

File: GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 Google
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m]: 2020

Comment (chliangGoogle) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m]: nit: 2018 -> 2020

File: GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+@interface GDTCORPlatformTest : GDTCORTestCase
+
+@end
+
+@implementation GDTCORPlatformTest
+
+/** Tests the reachability of mobile network connection in current platform. */
+- (void)testMobileConnectionReachability {
+  SCNetworkReachabilityFlags reachabilityFlags = [GDTCORReachability currentFlags];
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m]: wrap `[GDTCORReachability currentFlags]` in XCTAssertNoThrow and then do something like XCTAssertNotEqual(reachabilityFlags, 0) afterwards.

File: GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  XCTAssertNoThrow(networkConnectionType = GDTCORNetworkTypeMessage());
+  // The network connection type should be always WIFI in simulator logic test.
+  XCTAssertEqual(networkConnectionType, GDTCORNetworkTypeWIFI);
+}
+
+/** Tests mobile network connection subtype generating in current platform. */
+- (void)testGetNetworkMobileSubtype {
+  NSInteger networkMobileSubtype;
+  XCTAssertNoThrow(networkMobileSubtype = GDTCORNetworkMobileSubTypeMessage());
+  // The network connection type should be always UNKNOWN in simulator logic test.
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m]: s/type/mobile subtype/

File: GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+/** Tests background task creating of GDTCORApplication. */
+- (void)testGDTCORApplicationBeginBackgroundTask {
+  GDTCORApplication *application;
+  application = [[GDTCORApplication alloc] init];
+  __block GDTCORBackgroundIdentifier bgID;
+  XCTAssertNoThrow(bgID = [application beginBackgroundTaskWithName:@"GDTCORPlatformTest"
+                                                 expirationHandler:^{
+                                                   [application endBackgroundTask:bgID];
+                                                   bgID = GDTCORBackgroundIdentifierInvalid;
+                                                 }]);
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m]: Make sure to add an equivalent endBackgroundTask call for bgID, wrap that in a NoThrow

File: GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m [hunk 5]
```diff
@@ -68,9 +70,9 @@ - (void)testGDTCORApplicationBeginBackgroundTask {
   GDTCORApplication *application;
   application = [[GDTCORApplication alloc] init];
   __block GDTCORBackgroundIdentifier bgID;
+  XCTAssertNoThrow([application endBackgroundTask:bgID]);
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m]: this needs to go after the line where you start the background task

**Pull request merged**